pkg/agent/proxy/metrics/metrics.go

```diff
@@ -0,0 +1,117 @@
+package metrics
+
+import (
+	"sync"
+
+	kmetrics "k8s.io/component-base/metrics"
+	"k8s.io/component-base/metrics/legacyregistry"
+	"k8s.io/klog"
+)
+
+var (
+	once sync.Once
+
+	subsystemProxy = "antrea_proxy"
+
+	SyncProxyLatency = kmetrics.NewHistogram(
+		&kmetrics.HistogramOpts{
+			Subsystem:   subsystemProxy,
+			ConstLabels: map[string]string{"ipFamily": "v4"},
+			Name:        "sync_proxy_rules_duration_seconds",
+			Help:        "SyncProxyRules latency in seconds",
+			Buckets:     kmetrics.ExponentialBuckets(0.001, 2, 15),
+		},
+	)
+	ServicesInstalledTotal = kmetrics.NewGauge(
+		&kmetrics.GaugeOpts{
+			Subsystem:   subsystemProxy,
+			ConstLabels: map[string]string{"ipFamily": "v4"},
+			Name:        "services_installed_total",
+			Help:        "Current Services installed number",
+		},
+	)
+	EndpointsInstalledTotal = kmetrics.NewGauge(
+		&kmetrics.GaugeOpts{
+			Subsystem:   subsystemProxy,
+			ConstLabels: map[string]string{"ipFamily": "v4"},
+			Name:        "endpoints_installed_total",
+			Help:        "Current Endpoints installed number",
+		},
+	)
+	ServicesUpdatesTotal = kmetrics.NewCounter(
+		&kmetrics.CounterOpts{
+			Subsystem:   subsystemProxy,
+			ConstLabels: map[string]string{"ipFamily": "v4"},
+			Name:        "services_update_total",
+			Help:        "Cumulative Services updates times",
+		},
+	)
+	EndpointsUpdatesTotal = kmetrics.NewGauge(
+		&kmetrics.GaugeOpts{
+			Subsystem:   subsystemProxy,
+			ConstLabels: map[string]string{"ipFamily": "v4"},
+			Name:        "endpoints_update_total",
+			Help:        "Cumulative Endpoints updates times",
+		},
+	)
+
+	SyncProxyLatencyV6 = kmetrics.NewHistogram(
+		&kmetrics.HistogramOpts{
+			Subsystem:   subsystemProxy,
+			ConstLabels: map[string]string{"ipFamily": "v6"},
+			Name:        "sync_proxy_rules_duration_seconds",
+			Help:        "SyncProxyRules latency in seconds",
+			Buckets:     kmetrics.ExponentialBuckets(0.001, 2, 15),
+		},
+	)
+	ServicesInstalledTotalV6 = kmetrics.NewGauge(
+		&kmetrics.GaugeOpts{
+			Subsystem:   subsystemProxy,
+			ConstLabels: map[string]string{"ipFamily": "v6"},
+			Name:        "services_installed_total",
+			Help:        "Current Services installed number",
+		},
+	)
+	EndpointsInstalledTotalV6 = kmetrics.NewGauge(
+		&kmetrics.GaugeOpts{
+			Subsystem:   subsystemProxy,
+			ConstLabels: map[string]string{"ipFamily": "v6"},
+			Name:        "endpoints_installed_total",
+			Help:        "Current Endpoints installed number",
+		},
+	)
+	ServicesUpdatesTotalV6 = kmetrics.NewCounter(
+		&kmetrics.CounterOpts{
+			Subsystem:   subsystemProxy,
+			ConstLabels: map[string]string{"ipFamily": "v6"},
+			Name:        "services_update_total",
```

Comment:
suggestion: total_service_updates
Similarly for endpoints.

Reply:
Updated.